### ISSUE TITLE
Incorrect url resource resolution after library expansion

### ISSUE THREAD

**Issue by agustinlg** [raml-org/raml-js-parser-2#698]
We are having issues when parsing from a http url instead of a file.
This ONLY happens when calling the `toJson` AFTER `expand(true)`.

There actually seem to be two problems:
- It's calling the `httpResolver.getResource` even thought the `httpResolver.getResourceAsync` is defined.
- It's calling with an incorrectly build url (maybe it gets broken after the expand?)

This is the project that has the problem: [[DOC]_Referential_API-v1.zip](https://github.com/raml-org/raml-js-parser-2/files/843183/DOC._Referential_API-v1.zip), it's publicly hosted at: https://qax.anypoint.mulesoft.com/apiplatform/repository/v2/organizations/a65710f1-acac-4c93-9367-b1b4e41c3e30/public/apis/67158/versions/73464/files/root

Here is a small nodejs script to reproduce it, using that url:
```
const parser = require('raml-1-parser')
const request = require('request')

const url = 'https://qax.anypoint.mulesoft.com/apiplatform/repository/v2/organizations/a65710f1-acac-4c93-9367-b1b4e41c3e30/public/apis/67158/versions/73464/files/root';
parser.loadApi(url, {
    attributeDefaults: false,
    rejectOnErrors: true,
    httpResolver: {
      getResource: (url) => {
        throw new Error('Invalid call to httpResolver.getResource, with incorrect url: "' + url + '"')
      },
      getResourceAsync: (url) => {
        return new Promise((resolve, reject) => {
          request(url, (error, response, body) => {
            if (!error && response.statusCode == 200) {
              resolve({content: body})
            } else {
              reject(error || new Error('Could not fetch remote URL.'))
            }
          });
        });
      }
    }
  }
)
.then((api) => {
  console.log('JSON result OK:', api.expand(true).toJSON();
})
.catch((err) => {
  console.error(err)
})
```

Small note on the resolution of files from the API. There are just two services involved:
- The `/files/root` that returns the root raml file
- The `/files/{file_path}` that returns any other file by its path
At some point after the expand, it seems that paths like `../types/ContractsLib.raml` get incorreclty resolved, and so it calls a wrong path, removing the `/files` from the url.

**Comment by ddenisenko** [comment]
Fixed. Fix will be available in 1.1.18

**Comment by agustinlg** [comment]
Thanks!
